### PR TITLE
improve ariadne DB download

### PR DESCRIPTION
**Pull request by lindnemi**
- Ariadne DB needs to be used at several points in the workflow: for exogenous inputs and for evaluation
- The values pulled from the DB are scenario dependent
- The exogenous inputs should not be based on the "Hybrid" Modell, but on the "Leitmodelle"

This PR:

Downloads all Leitmodelle and the most commonly used Scenarios at the beginning of the workflow.
All downstream scripts, should use the the downloaded DB.

Open points in relation to Ariadne DB:

- CO2-Targets: The targets need to be specified per scenario. The scenario is defined in the config. The CO2-Script should modify the config. Hence the co2 script both depends and modifies the config. How to break this circle?
- Modify the Export PR accordingly

### CODE REVIEW

File: workflow/scripts/retrieve_ariadne_database.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        model=[
+            # "Hybrid", # Download only the Leitmodelle
+            "REMIND-EU v1.1", 
+            'REMod v1.0', 
+            'TIMES PanEU v1.0', 
+            'FORECAST v1.0',
+            'DEMO v1',
+        ],
+        scenario=[# Download only the most recent iterations of scenarios
+            "8Gt_Bal_v3",
```
Comment (toniseibold) [inline workflow/scripts/retrieve_ariadne_database.py:27]: Could we do that with data from the config.yaml file? Seems to be tedious to change it every time a new version is published in this script

File: workflow/scripts/retrieve_ariadne_database.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        ],
+        scenario=[# Download only the most recent iterations of scenarios
+            "8Gt_Bal_v3", 
+            "8Gt_Elec_v3", 
+            "8Gt_H2_v3",
+        ],
+    )
+
+    db.timeseries().to_csv(snakemake.output.data)
+
```
Comment (toniseibold) [inline workflow/scripts/retrieve_ariadne_database.py:34]: Maybe delete those extra lines at the end of the script

File: workflow/Snakefile [hunk 3]
```diff
@@ -84,18 +84,19 @@ rule clean:
          print("Data downloaded to data/ has not been cleaned.")
 
 
-rule retrieve_ariadne_scenario_data:
+rule retrieve_ariadne_database:
     params:
         iiasa_usr=os.environ["IIASA_USERNAME"],
```
Comment (toniseibold) [inline workflow/Snakefile:89]: I would add [iiasa_database][leitmodelle] and [iiasa_databse][scenarios] as parameters from the config.yaml

File: workflow/scripts/retrieve_ariadne_database.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        snakemake = mock_snakemake("retrieve_ariadne_database")
+
+    pyam.iiasa.set_config(snakemake.params.iiasa_usr, snakemake.params.iiasa_pwd)
+
+    db = pyam.read_iiasa(
+        "ariadne_intern",
+        model=[
+            # "Hybrid", # Download only the Leitmodelle
+            "REMIND-EU v1.1",
```
Comment (toniseibold) [inline workflow/scripts/retrieve_ariadne_database.py:20]: Same as below

File: workflow/scripts/build_scenarios.py [hunk 5]
```diff
@@ -5,39 +5,44 @@
...
-import pyam
+
 import ruamel.yaml
 from pathlib import Path
 import pandas as pd
 import os
 
-def get_shares(df):
+def get_shares(df, planning_horizons):
     # Get share of vehicles for transport sector - meglecting heavy duty vehicles
```
Comment (toniseibold) [inline workflow/scripts/build_scenarios.py:15]: typo

File: workflow/scripts/build_scenarios.py [hunk 6]
```diff
@@ -5,39 +5,44 @@
...
-    total_transport = df.loc["Stock|Transportation|LDV"]
-    tech_transport = df.loc[["Stock|Transportation|LDV|ICE",
-                "Stock|Transportation|LDV|BEV",
-                "Stock|Transportation|LDV|PHEV",]]
+    # At the
```
Comment (toniseibold) [inline workflow/scripts/build_scenarios.py:16]: ???

File: workflow/scripts/build_scenarios.py [hunk 7]
```diff
@@ -83,21 +88,26 @@ def write_to_scenario_yaml(output, scenarios, transport_share, naval_share):
         snakemake = mock_snakemake("build_scenarios")
 
     # Set USERNAME and PASSWORD for the Ariadne DB
-    pyam.iiasa.set_config(
-        snakemake.params.iiasa_usr, 
-        snakemake.params.iiasa_pwd,
+    ariadne = pd.read_csv(
```
Comment (toniseibold) [inline workflow/scripts/build_scenarios.py:91]: variable name `ariadne` is a bit ambigous

File: workflow/scripts/build_scenarios.py [hunk 8]
```diff
@@ -83,21 +88,26 @@ def write_to_scenario_yaml(output, scenarios, transport_share, naval_share):
...
-    transport_share, naval_share = get_shares(df)
+    planning_horizons = [2020, 2025, 2030, 2035, 2040, 2045]
+    transport_share, naval_share = get_shares(df, planning_horizons)
 
     scenarios = snakemake.params.scenario_name
-    output = snakemake.input[0]
-    write_to_scenario_yaml(output, scenarios, transport_share, naval_share)
\ No newline at end of file
+
+    if "snakemake" in globals():
```
Comment (toniseibold) [inline workflow/scripts/build_scenarios.py:108]: Why? with mocksnakemake I find the same path with `snakemake.input.scenario_yaml`

File: workflow/scripts/build_scenarios.py [hunk 9]
```diff
@@ -5,39 +5,44 @@
...
     transport_share.set_index(pd.Index(["ICE", "BEV", "PHEV"]), inplace=True)
 
     # Get share of Navigation fuels
-    total_navigation = df.loc[["Final Energy|Bunkers|Navigation",
-                         "Final Energy|Transportation|Domestic Navigation"]].sum(axis=0)
-    navigation_liquid = df.loc[["Final Energy|Bunkers|Navigation",
-                         "Final Energy|Transportation|Domestic Navigation|Liquids"]].sum()
-    navigation_h2 = df.loc[["Final Energy|Transportation|Domestic Navigation|Hydrogen"]]    
+    total_navigation = \
+        df.loc["REMIND-EU v1.1", "Final Energy|Bunkers|Navigation"] + \
```
Comment (toniseibold) [inline workflow/scripts/build_scenarios.py:30]: Is there a reason why you use two different models for that? If so maybe leave a comment

File: config/config.yaml [hunk 10]
```diff
@@ -19,8 +19,17 @@ run:
...
+    - REMIND-EU v1.1
+    - REMod v1.0
+    - TIMES PanEU v1.0
+    - FORECAST v1.0
+    - DEMO v1
+  scenarios: 
+    - 8Gt_Bal_v3
+    - 8Gt_Elec_v3
+    - 8Gt_H2_v3
+  reference_scenario: 8Gt_Bal_v3
```
Comment (toniseibold) [inline config/config.yaml:32]: Is there a reason for defining the `config["iiasa_database"]["reference_scenario"]` here?

File: workflow/Snakefile [hunk 11]
```diff
@@ -229,13 +230,14 @@ if config["wasserstoff_kernnetz"]["enable"]:
...
-        iiasa_scenario=config["iiasa_database"]["scenario"],
+        iiasa_scenario=config["iiasa_database"]["reference_scenario"],
         scenario_name=config["run"]["name"],
     input:
-        "config/scenarios.yaml"
+        #default_scenarios_yaml="config/scenarios.default.yaml",
+        ariadne_database=resources("ariadne_database.csv"),
+        scenario_yaml=config["run"]["scenarios"]["file"],
+    #output:
+    #    config["run"]["scenarios"]["file"],
```
Comment (toniseibold) [inline workflow/Snakefile:240]: maybe delete those commented lines

File: workflow/scripts/modify_existing_heating.py [hunk 12]
```diff
@@ -2,30 +2,52 @@
...
-                  "oil boiler" : "Oil Boiler",
-                  "air heat pump" : "Heat Pump|Electrical|Air",
-                  "ground heat pump" : "Heat Pump|Electrical|Ground",
-                  "biomass boiler" : "Biomass Boiler"}
+ariadne = pd.read_csv(
+    snakemake.input.ariadne,
+    index_col=["model", "scenario", "region", "variable", "unit"]
+).loc[
+    leitmodell,
+    snakemake.config["iiasa_database"]["scenario"],
```
Comment (toniseibold) [inline workflow/scripts/modify_existing_heating.py:15]: bug should be "reference_scenario"

**Review by toniseibold — CHANGES_REQUESTED** [review]

File: config/config.yaml (hunk 10 above)
Comment (lindnemi) [reply config/config.yaml:32]: fallback, may change later

File: workflow/scripts/modify_existing_heating.py (hunk 12 above)
Comment (lindnemi) [reply workflow/scripts/modify_existing_heating.py]: good catch!

**Review by toniseibold — APPROVED** [review]

**Pull request merged**